src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -166,6 +158,12 @@
         type="udf_body",
         trim_chars=("$",),
     ),
+    StagePath=RegexParser(
+        r"(?:@[^\s;)]+|'@[^']+')",
+        CodeSegment,
+        name="stage_path",
```

Comment:
I see you've also added a `RegexLexer` with the same name. Are both of these necessary? (I haven't worked on dialects in a while, so I could be forgetting something.)

Reply:
Yes, based on my testing while writing this PR, I think so. However, this is my first time touching the lexers, so if there is a more efficient way to implement the stage paths, I would be grateful for any hint!

Reply:
None from me, haha. I have mostly been working on core stuff for the past few months, so I've forgotten the details. From what I remember, though, this is 👍 .